kale/embed/factorization.py

```diff
@@ -267,80 +269,123 @@ def inverse_transform(self, x):
+        kernel (str): Kernel to use for MIDA. Defaults to "linear".
+        gamma (float): Kernel coefficient for rbf, poly and sigmoid kernels. Ignored by other kernels. If ``gamma`` is
+            ``None``, then it is set to ``1/n_features``., default=None.
+        degree (float): Degree for poly kernels. Ignored by other kernels., default=3
+        coef0 (float): Independent term in poly and sigmoid kernels.Ignored by other kernels. Default=1
         mu (float): Hyperparameter of the l2 penalty. Defaults to 1.0.
         eta (float): Hyperparameter of the label dependence. Defaults to 1.0.
-        augmentation (bool): Whether using covariates as augment features. Defaults to False.
+        augmentation (bool): Whether using domain/group labels as augment features. Defaults to False.
+        fit_label (bool): Whether to fit the label for semi-supervised MIDA. Defaults to False.
         kernel_params (dict or None): Parameters for the kernel. Defaults to None.
 
     References:
         [1] Yan, K., Kou, L. and Zhang, D., 2018. Learning domain-invariant subspace using domain features and
             independence maximization. IEEE transactions on cybernetics, 48(1), pp.288-299.
     """
 
+    _parameter_constraints: dict = {
+        "n_components": [
+            Interval(Integral, 1, None, closed="left"),
+            None,
+        ],
+        "kernel": [
+            StrOptions({"linear", "poly", "rbf", "sigmoid", "cosine", "precomputed"}),
+            callable,
+        ],
+        "gamma": [
+            Interval(Real, 0, None, closed="left"),
+            None,
+        ],
+        "degree": [Interval(Real, 0, None, closed="left")],
+        "coef0": [Interval(Real, None, None, closed="neither")],
+        "kernel_params": [dict, None],
+        "mu": [Real],
+        "eta": [Real],
+        "fit_label": ["boolean"],
+        "augmentation": ["boolean"],
+        "n_jobs": [None, Integral],
+    }
+
     def __init__(
         self,
-        n_components,
+        n_components=None,
         kernel="linear",
-        lambda_=1.0,
         mu=1.0,
         eta=1.0,
+        gamma=None,
+        degree=3,
+        coef0=1,
+        fit_label=False,
         augmentation=False,
         kernel_params=None,
+        n_jobs=None,
     ):
         self.n_components = n_components
         self.kernel = kernel
+        self.gamma = gamma
+        self.degree = degree
+        self.coef0 = coef0
         self.mu = mu
         self.eta = eta
         self.augmentation = augmentation
-        if kernel_params is None:
-            self.kernel_params = {}
-        else:
-            self.kernel_params = kernel_params
-        self._label_binarizer = LabelBinarizer(pos_label=1, neg_label=-1)
-        self._centerer = KernelCenterer()
+        self.kernel_params = kernel_params
+        self.fit_label = fit_label
         self.x_fit = None
+        self._label_binarizer = None
+        self._centerer = None
+        self._onehot = None
```

Comment:
Likewise for `_label_binarizer`, `_centerer`, and `_onehot`, we only initialize it during `fit`. And use standard definition of attributes initialized during `fit` such as `centerer_`. I would recommend these to be moved to avoid messing with `BaseEstimator.set_params(...)`